gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/apim/core/integration/use_case/IngestIntegrationApisUseCase.java

```diff
@@ -103,11 +109,40 @@ public Completable execute(Input input) {
                         if (api.plans() != null) {
                             createPlans(api.plans(), federatedApi, auditInfo);
                         }
+
+                        if (api.pages() != null) {
+                            createDocumentation(api, federatedApi.getId());
+                        }
                     });
             })
             .ignoreElements();
     }
 
+    private void createDocumentation(IntegrationApi integrationApi, String referenceId) {
+        var now = Date.from(TimeProvider.instantNow());
+        integrationApi
+            .pages()
+            .stream()
+            .map(page ->
+                Page
+                    .builder()
+                    .id(UuidString.generateRandom())
+                    .name(integrationApi.name().concat("-oas.yml"))
+                    .content(page.content())
+                    .type(Page.Type.valueOf(page.pageType().name()))
+                    .referenceId(referenceId)
+                    .referenceType(Page.ReferenceType.API)
+                    .published(true)
+                    .visibility(Page.Visibility.PRIVATE)
+                    .homepage(true)
+                    .configuration(Map.of("tryIt", "true", "viewer", "Swagger"))
+                    .createdAt(now)
+                    .updatedAt(now)
+                    .build()
+            )
```

Comment:
The mapping is not that easy I think. Here it is working because we only have a SWAGGER page. I think we should apply a mapping per type of page (and only focus on `SWAGGER` type for now) 
With that code, if the agent send a `Markdown` page, we will create a page named `[apiname]-oas.yml` which is incorrect because it should be `.md` file 🤔